src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseApprovalService.cs

```diff
@@ -62,19 +62,31 @@ public class ReleaseApprovalService : IReleaseApprovalService
                         .ThenInclude(rs => rs.CreatedBy))
                 .OnSuccess(_userService.CheckCanViewReleaseStatusHistory)
                 .OnSuccess(release =>
-                    release.ReleaseStatuses
+                {
+                    var allReleaseVersionIds = _context.Releases
+                        .Where(r => r.PublicationId == release.PublicationId
+                                    && r.ReleaseName == release.ReleaseName
+                                    && r.TimePeriodCoverage == release.TimePeriodCoverage)
+                        .Select(r => r.Id)
+                        .ToList();
```

Comment:
I see why it's preferable to do this rather than iterating over the Publications releases because we think the overhead of doing so will be high, but just wondering what the difference would be?

I'm not really sure about this because there's no relationship between versions of a Release except for `PreviousVersionId`, so I would say this just happens to work 'ok' on the basis that analysts have been unlikely to change `Year` (ReleaseName) or `TimePeriodCoverage` between amendments in the past and same for in the future, rather than being the safest way of querying them.

Doesn't a bug exist here where by changing the release Year or Time identifier on an amendment, you end up missing the history of other amendments plus the history of the original version?

Similarly a bug where if you amend a different release of the same publication and change both the Year and Time Identifier to match another release, you effectively merge the status history into that unrelated release?

The data in Prod  looks like altering Year and Time Identifier in this has never happened but there are examples in other environments.

Two types of examples:
* Where the Year/Time identifier has changed between amendments of releases related to each other by previous version.
* Unrelated amendments within the same publication that use the the same Year and Time identifier.

If analysts shouldn't be changing these fields on amendments then maybe we should prevent it to stop the possibility of missing or incorrect status history in the future, given that it looks like it hasn't occurred in Prod yet, but even then getting them by PreviousVersionId is the safe way.